packaging/elasticsearch_plugin.py

```diff
@@ -111,30 +115,54 @@ def parse_plugin_repo(string):
 
     return repo
 
-
 def is_plugin_present(plugin_dir, working_dir):
     return os.path.isdir(os.path.join(working_dir, plugin_dir))
 
-
 def parse_error(string):
     reason = "reason: "
     try:
         return string[string.index(reason) + len(reason):].strip()
     except ValueError:
         return string
 
+def install_plugin(module, plugin_bin, plugin_name, url, proxy_host, proxy_port, timeout):
+    cmd_args = [plugin_bin, PACKAGE_STATE_MAP["present"], plugin_name]
 
-def main():
+    if proxy_host and proxy_port:
+        cmd_args.append("-DproxyHost=%s -DproxyPort=%s" % (proxy_host, proxy_port))
 
-    package_state_map = dict(
-        present="install",
-        absent="remove"
-    )
+    if url:
+        cmd_args.append("--url %s" % url)
+
+    if timeout:
+        cmd_args.append("--timeout %s" % timeout)
```

Comment:
Maybe someone with Python knowledge can help me out here, but I had trouble using this code against an older version of Elasticsearch. Splitting the args into two works as such

``` python
cmd_args.append("--timeout")
cmd_args.append(timeout)
```

Not sure on _why_ this happens.